flask_test_app/app.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Flask application made for integration tests."""
+
+import time
+from urllib.parse import urlparse
+
+import psycopg
+import pymysql
+import pymysql.cursors
+from flask import Flask, g, jsonify, request
+
+app = Flask(__name__)
+app.config.from_prefixed_env()
+
+
+def get_mysql_database():
+    """Get the mysql db connection."""
+    if "mysql_db" not in g:
+        if app.config.get("MYSQL_DB_CONNECT_STRING"):
+            uri_parts = urlparse(app.config["MYSQL_DB_CONNECT_STRING"])
+            g.mysql_db = pymysql.connect(
+                host=uri_parts.hostname,
+                user=uri_parts.username,
+                password=uri_parts.password,
+                database=uri_parts.path[1:],
+                port=uri_parts.port,
+            )
+        else:
+            return None
+    return g.mysql_db
+
+
+def get_postgresql_database():
+    """Get the postgresql db connection."""
+    if "postgresql_db" not in g:
+        if app.config.get("POSTGRESQL_DB_CONNECT_STRING"):
+            uri_parts = urlparse(app.config.get("POSTGRESQL_DB_CONNECT_STRING"))
+            g.postgresql_db = psycopg.connect(
```

Comment:
Do these support passing in the connection string as well?

Reply:
Fixed